client/my-sites/email/gsuite-add-users/index.jsx

```diff
@@ -283,11 +285,13 @@ class GSuiteAddUsers extends React.Component {
 				{ selectedSite && <QueryGSuiteUsers siteId={ selectedSite.ID } /> }
 
 				<Main wideLayout={ true }>
-					<DocumentHead title={ translate( 'Add New Users' ) } />
+					<DocumentHead title={ translate( 'Add New Mailboxes' ) } />
 
 					<EmailHeader currentRoute={ currentRoute } selectedSite={ selectedSite } />
 
-					<HeaderCake onClick={ this.goToEmail }>{ translate( 'Add new users' ) }</HeaderCake>
+					<HeaderCake onClick={ this.goToEmail }>
+						{ googleMailServiceFamily + ': ' + selectedDomainName }
+					</HeaderCake>
```

Comment:
This is to be consistent with what we display in the header of the `Add New Mailboxes` page for Titan.